src/engine/idTable/CompressedExternalIdTable.h

```diff
@@ -478,27 +527,27 @@ class CompressedExternalIdTable
   // `IdTable in the order that they were `push`ed. This function may be called
   // exactly once.
   auto getRows() {
+    using RowType = typename IdTableStatic<NumStaticCols>::row_type;
     if (!this->transformAndPushLastBlock()) {
-      // For the case of only a single block we have to mimic the exact same
-      // return type as that of `writer_.getGeneratorForAllRows()`, that's why
-      // there are the seemingly redundant multiple calls to
-      // join(OwningView(vector(...))).
-      auto generator = [](const auto& block)
-          -> cppcoro::generator<const IdTableStatic<NumStaticCols>> {
-        co_yield block;
-      }(this->currentBlock_);
-      auto rowView =
-          ql::views::join(ad_utility::OwningView{std::move(generator)});
-      std::vector<decltype(rowView)> vec;
-      vec.push_back(std::move(rowView));
-      return ql::views::join(ad_utility::OwningViewNoConst(std::move(vec)));
+      auto rowGenerator = [block = std::move(this->currentBlock_),
+                           pos =
+                               size_t{0}]() mutable -> std::optional<RowType> {
+        if (pos < block.size()) {
+          return block[pos++];
+        }
+        return std::nullopt;
+      };
+      return ad_utility::InputRangeTypeErased<RowType>(
+          ad_utility::InputRangeFromGetCallable(std::move(rowGenerator)));
     }
     this->pushBlock(std::move(this->currentBlock_));
     this->resetCurrentBlock(false);
     if (this->compressAndWriteFuture_.valid()) {
       this->compressAndWriteFuture_.get();
     }
-    return this->writer_.template getGeneratorForAllRows<NumStaticCols>();
+    auto joined =
+        this->writer_.template getGeneratorForAllRows<NumStaticCols>();
+    return ad_utility::InputRangeTypeErased<RowType>(std::move(joined));
```

Comment:
Same here, should be abel to simply `return joined`, where `joind` is `join_view<TypeErase<IdTable>>`

Reply:
see above.